src/dbnode/storage/index.go

```diff
@@ -1017,6 +1054,7 @@ func (i *nsIndex) query(
 	i.state.RUnlock()
 
 	if err != nil {
+		sp.LogFields(opentracinglog.Error(err))
```

Comment:
Hm instead of logging error at every single place that returns an error, can you just create a sub function, and then log error if any of the subfunction returns an error?

Also makes it easy to avoid needing the defer.
i.e.
```go
func (i *nsIndex) query(
	ctx context.Context,
	query index.Query,
	results index.BaseResults,
	opts index.QueryOptions,
	execBlockFn execBlockQueryFn,
) (bool, error) {
	ctx, sp := ctx.StartTraceSpan(tracepoint.NSIdxSubQuery)
	sp.LogFields(
		opentracinglog.String("query", query.String()),
		opentracinglog.String("namespace", i.nsMetadata.ID().String()),
		opentracinglog.Int("limit", opts.Limit),
		xopentracing.Time("start", opts.StartInclusive),
		xopentracing.Time("end", opts.EndExclusive),
	)
	defer sp.Finish()

	result, err := i.queryWithSpan(ctx, query, results, opts, execBlockFn, sp)
	if err != nil {
		sp.LogFields(opentracinglog.Error(err))
	}

	return result, err
}

func (i *nsIndex) queryWithSpan(
	ctx context.Context,
	query index.Query,
	results index.BaseResults,
	opts index.QueryOptions,
	execBlockFn execBlockQueryFn,
	span opentracing.Span,
) (bool, error) {
  // .. all existing code
}
```

Same with every other large code paths with lots of early returns in this PR.

Reply:
Sounds good - will update.